config/management/src/validator_config.rs

```diff
@@ -103,8 +104,9 @@ pub fn build_validator_config_transaction<S: KVStorage + CryptoStorage>(
         .map(|v| v.value)?;
 
     let consensus_key = validator_storage
-        .get_public_key(CONSENSUS_KEY)
-        .map(|v| v.public_key)?;
+        .get::<bls12381::PrivateKey>(CONSENSUS_KEY)
+        .map(|v| v.value)?
+        .public_key();
```

Comment:
Are we concerned about the performance implications of having to convert the SK to a PK here? Or is this not called frequently enough (e.g., per block)?

Reply:
this is only used during genesis